cmd/contour/serve.go

```diff
@@ -335,6 +335,7 @@ func (s *Server) doServe() error {
 		Timeouts:                     timeouts,
 		DefaultHTTPVersions:          parseDefaultHTTPVersions(contourConfiguration.Envoy.DefaultHTTPVersions),
 		AllowChunkedLength:           !contourConfiguration.Envoy.Listener.DisableAllowChunkedLength,
+		MergeSlashes:                 contourConfiguration.Envoy.Listener.MergeSlashes,
```

Comment:
Assuming we switch to `DisableMergeSlashes`, this is where I'd apply the negation (just like the line just above for `DisableAllowChunkedLength`), so that the `xdscache_v3.ListenerConfig` still just has a `MergeSlashes` field.